ControllerLogic/DateTimeProvider.cs

```diff
@@ -0,0 +1,12 @@
+public class DateTimeProvider : IDateTimeProvider
+{
+    public DateTime GetCurrentDateTime()
+    {
+        return DateTime.Now;
+    }
+
+    public long GetCurrentMiliseconds()
+    {
+        return (long)(DateTime.Now - new DateTime(1970, 1, 1)).TotalMilliseconds;
```

Comment:
new DateTime(1970, 1, 1) could be declared once as private readonly static field

Reply:
Ogólnie jeżeli jest jedno linijkowiec to użyjmy body expression bo wsm ładniej to wygląda:

public DateTime GetCurrentDateTime() => DateTime.Now;